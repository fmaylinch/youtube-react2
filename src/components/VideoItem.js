import React from 'react';

const VideoItem = (props) => {

    console.log("props received", props)

    return (
        <div> 
            <h1>{props.title}</h1>
            <iframe 
                title={props.title}
                width="320" height="180"
                src={"https://www.youtube.com/embed/" + props.videoId}
                frameBorder="0" allowFullScreen>
            </iframe>
        </div>
    )
}
    
export default VideoItem;
