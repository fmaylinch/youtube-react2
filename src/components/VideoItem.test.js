//@flow

import React from 'react'
import { shallow } from 'enzyme'
import VideoItem from './VideoItem'
import type { Video } from '../components/types.js'

describe('VideoItem', () => {

    const video = {
        id: "4TPqUvy1vYU",
        image: "https://i.ytimg.com/vi/4TPqUvy1vYU/mqdefault.jpg",
        title: "St. Vincent - New York (Official Video)"
    }

    it('renders correctly', () => {
        const component = shallow(<VideoItem video={video}/>)
        expect(component).toMatchSnapshot()
    })

    it('renders image correctly', () => {
        const component = shallow(<VideoItem video={video}/>)
		const image = <img src={video.image} alt={video.title} />
		expect(component.contains(image)).toBeTruthy()
    })
})