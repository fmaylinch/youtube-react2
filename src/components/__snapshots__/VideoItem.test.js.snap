// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem renders correctly 1`] = `
<div>
  <img
    alt="St. Vincent - New York (Official Video)"
    src="https://i.ytimg.com/vi/4TPqUvy1vYU/mqdefault.jpg"
  />
</div>
`;
